### PR TITLE
Fix colour of avatar and colour matching with username

### PR DESCRIPTION
**Pull request by germain-gg**
Fixes https://github.com/vector-im/element-web/issues/26042

## Checklist

-   [x] Tests written for new code (and old code if feasible)
-   [x] Linter and other CI checks pass
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Fix colour of avatar and colour matching with username ([\#11470](https://github.com/matrix-org/matrix-react-sdk/pull/11470)). Fixes vector-im/element-web#26042.

### CODE REVIEW

File: src/components/views/messages/DisambiguatedProfile.tsx [hunk 1]
```diff
@@ -40,7 +40,7 @@ export default class DisambiguatedProfile extends React.Component<IProps> {
 
         let colorClass: string | undefined;
         if (colored) {
-            colorClass = getUserNameColorClass(fallbackName);
+            colorClass = getUserNameColorClass(mxid ?? "");
```
Comment (andybalaam) [inline src/components/views/messages/DisambiguatedProfile.tsx:43]: Is there a test for this? I'd love to see one if not. Happy for you to merge and add it after if needed though.

File: src/utils/FormattingUtils.ts [hunk 2]
```diff
@@ -73,30 +74,11 @@ export function formatBytes(bytes: number, decimals = 2): string {
...
-        hash |= 0;
-    }
-    return Math.abs(hash);
-}
 
-export function getUserNameColorClass(userId?: string): string {
-    const colorNumber = (hashCode(userId) % 8) + 1;
-    return `mx_Username_color${colorNumber}`;
+export function getUserNameColorClass(userId: string): string {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
```
Comment (andybalaam) [inline src/utils/FormattingUtils.ts:79]: Can I have a comment explaining why this is OK to disable?

Comment (andybalaam) [reply src/utils/FormattingUtils.ts:79]: Same comment about a test as above.

Comment (germain-gg) [reply src/utils/FormattingUtils.ts:79]: The function starts with `use` but isn't actually a hook. The underlying class is tested within Compound but not here. I might land this to fix the dev build and get back to it

File: src/components/views/messages/DisambiguatedProfile.tsx (hunk 1 above)
Comment (germain-gg) [reply src/components/views/messages/DisambiguatedProfile.tsx:43]: I will do that! thank you 🙏

File: src/utils/FormattingUtils.ts (hunk 2 above)
Comment (andybalaam) [reply src/utils/FormattingUtils.ts:79]: Great, can we have that comment in the code please?

Comment (t3chguy) [reply src/utils/FormattingUtils.ts:79]: We should avoid the naming as react will apply hook behaviours to it if called from functional components which adds performance overhead.

**Pull request merged**